picamera2/devices/imx500/imx500.py

```diff
@@ -45,6 +47,177 @@ class CnnOutputTensorInfoExported(ctypes.LittleEndianStructure):
         ('info', OutputTensorInfo * MAX_NUM_TENSORS)
     ]
 
+class NetworkIntrinsics:
+    def __init__(self, val=None):
+        self.__intrinsics: Optional[dict] = None
+        self.__schema = {
+            "$schema": "https://json-schema.org/draft-07/schema",
+            "title": "network_intrinsics",
+            "type": "object",
+            "properties": {
+                "task": {
+                    "type": "string",
+                    "enum": ["classification", "object detection", "segmentation"],
+                    "description": "Network task",
+                },
+                "inference_rate": {"type": "number", "minimum": 0},
+                "cpu": {
+                    "type": "object",
+                    "properties": {
+                        "bbox_normalization": {"type": "boolean"},
+                        "softmax": {"type": "boolean"},
+                        "post_processing": {"type": "string"},
+                    },
+                },
+                "input_aspect_ratio": {
+                    "type": "object",
+                    "properties": {
+                        "width": {"type": "integer", "exclusiveMinimum": 0},
+                        "height": {"type": "integer", "exclusiveMinimum": 0},
+                    },
+                    "required": ["width", "height"],
+                },
+                "classes": {
+                    "type": "object",
+                    "properties": {
+                        "labels": {"type": "array", "items": {"type": "string"}},
+                        "ignore_undefined": {"type": "boolean"},
+                    },
+                },
+            },
+        }
+        if val is not None:
+            jsonschema.validate(val, self.__schema)
+            self.__intrinsics = val
+
+    @property
+    def intrinsics(self) -> Optional[dict]:
+        return self.__intrinsics
+
+    @intrinsics.setter
+    def intrinsics(self, val):
+        jsonschema.validate(val, self.__schema)
+        self.__intrinsics = val
+
+    def __repr__(self):
+        return json.dumps(self.__intrinsics) if self.__intrinsics else ""
```

Comment:
Do we want it to look pretty with ``json.dumps(self.__intrinsics, indent=4)``?  I'm not really fussed.